tests/client/test_client.py

```diff
@@ -33,33 +34,44 @@ def test_post_json(server):
     assert response.reason_phrase == "OK"
 
 
+def test_stream_manger_release_resources(server):
+    with httpx.Client() as http:
+        response = http.post(
+            server.url.copy_with(path="/echo_body"), stream=True, data=b"*" * 409600
+        )
+
+    assert response.status_code == 200
+    with pytest.raises(RemoteProtocolError):
+        response.read()
```

Comment:
I think we should not let an `h11` exception bubble up to our users; we need to raise an HTTPX-specific exception here. I believe `ResponseClosed` is the most adequate.

This made me realize that we are not force-closing responses yet here on client exit. You can see this by turning on `HTTPX_DEBUG` and seeing `receive_event NEED_DATA` log lines because the response object is still trying to receive response data, and that is because we did not call `.close()` on it to send a `ResponseClosed` event to h11.

I guess implementing response closing on client exit involves some form of storing "pending responses" on the client. We'd add the response there in the `.request()` methods, remove it when the response's `on_close()` callback is normally called, and iterate over it in client's `__exit__()` or `__aexit__()` to close any pending responses.

Would you like to look into this?